src/compress.rs

```diff
@@ -146,11 +148,40 @@ mod snappy {
 		}
 
 		pub(super) fn compress(&self, value: &[u8]) -> Vec<u8> {
-			snappy::compress(value)
+			let mut buf = Vec::with_capacity(value.len() << 3);
+			let mut encoder = snap::write::FrameEncoder::new(&mut buf);
+			encoder.write(value).expect("Expect in memory write to succeed.");
```

Comment:
```suggestion
			let mut encoder = snap::write::FrameEncoder::new(&mut buf);
			encoder.write(value).expect("Expect in memory write to succeed.");
			std::mem::drop(encoder);
```

Reply:
I had to drop the encoder when using your code. (using snap 1.0.5 and rustc 1.52.1)